File: micro-digiservices/ecreditservice/src/main/java/io/digiservices/ecreditservice/dto/CreateStockDto.java
package io.digiservices.ecreditservice.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateStockDto {
    @NotNull(message = "L'ID utilisateur est obligatoire")
    private Long idUser;

    @NotBlank(message = "Le service est obligatoire")
    private String service;

    private String detailBonCommande;
    private Integer qte;
    private Long pointventeId;
    private Long agenceId;
    private Long delegationId;
    private Long categorieId;
    private String observations;
}

File: frontend/ultima-ng-19.0.0/src/app/interface/CreateStockDto.ts
export interface CreateStockDto {
    idUser: number;
    service: string;
    detailBonCommande?: string;
    pointventeId?: number;
    agenceId?: number;
    delegationId?: number;
    categorieId?: number;
    observations?: string;
    qte?: number;
}
